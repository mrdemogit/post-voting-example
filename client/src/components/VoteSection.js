import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import UpvoteIcon from './ArrowIcon'

const StyledButton = styled.button`
  font-size: 1em;
  margin: 0.3em;
  padding: 0.1em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  color: #338cb1;
  border-color: #fff;
  cursor: pointer;
`

const StyledButtonUp = styled(StyledButton)`
  transform: rotate(180deg);
  transform-origin: 50% 50%;
`

const VoteSection = ({ uuid, votes, upvote, downvote }) => {
  return (
    <Fragment>
      <StyledButtonUp onClick={() => upvote(uuid)}>
        <UpvoteIcon />
      </StyledButtonUp>
      <StyledButton onClick={() => downvote(uuid)}>
        <UpvoteIcon />
      </StyledButton>
      <span>{votes}</span>
    </Fragment>
  )
}

VoteSection.propTypes = {
  uuid: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired
}

export default VoteSection
