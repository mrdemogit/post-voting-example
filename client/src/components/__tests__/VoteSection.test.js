import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import VoteSection from '../VoteSection'

describe('VoteSection', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <VoteSection
        uuid='uuid1'
        votes={123}
        upvote={jest.fn()}
        downvote={jest.fn()}
      />
    )

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
