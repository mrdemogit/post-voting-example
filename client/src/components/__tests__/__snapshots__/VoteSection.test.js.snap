// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteSection should render correctly 1`] = `
<React.Fragment>
  <ForwardRef
    onClick={[Function]}
  >
    <Arrow />
  </ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    <Arrow />
  </ForwardRef>
  <span>
    123
  </span>
</React.Fragment>
`;
